### PR TITLE
Fix highlight for unquoted attributes

### LINKED ISSUE #46: Highlighting attributes breaks when not wrapped with "
highlighting-error

According to the HTML specifications, attributes do not require to be wrapped in ".

### PR DESCRIPTION
**Pull request by csvn**
The tests I have done seem to indicate this works correctly. I also used "Inspect TM scopes" to check that attribute names and tokens still had correct scopes, which was true as far as I could see.

image

image

I was not able to find any documentation on the syntax for `injectionSelector`, so `meta.tag` scope was the best I could find that still matches native tags, custom tags, void/regular tags etc. I tried to use `(meta.tag.html)` and `(meta.tag.*.html)` to limit the selector, but that did not work. These were the scopes I could find that should be targeted:

- `meta.tag.custom.start.html`
- `meta.tag.structure.<tagname>.<start|void>.html`
- `meta.tag.other.<tagname>.html`

I'm not sure if there are performance issues with using only `meta.tag`, or if it is too general, but any tips would be helpful.

Fixes #46

### CODE REVIEW

File: syntaxes/lit-html-string-injection.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+	"fileTypes": [],
+	"injectionSelector": "L:source.js (meta.tag), L:source.jsx (meta.tag), L:source.js.jsx (meta.tag), L:source.ts (meta.tag), L:source.tsx (meta.tag)",
+	"patterns": [
```
Comment (mjbvz) [inline syntaxes/lit-html-string-injection.json]: This approach looks good. My only concerns is that `meta.tag` is alway used by jsx, so this selector may effect react code too

I think a safer selector would be:

```
"injectionSelector": "L:meta.embedded.block.html meta.tag",
```

Can you try that to see if works

Comment (csvn) [reply syntaxes/lit-html-string-injection.json]: Good idea, and thanks for the suggestion!

I tested on `highlighting.test.js` and a local Typescript project, which both seems to work well. Unfortunately I don't have any handy JSX projects to look at, but the `meta.embedded.block.html` should only be applied for `html`` ` tagged template string literals if I understand the `lit-html.json` file correctly.

File: package.json [hunk 2]
```diff
@@ -90,8 +90,8 @@
                     "source.tsx",
                     "text.html.basic"
```
Comment (csvn) [inline package.json]: I'm not too familiar with how this works, but should this `injectTo` field maybe be changed to `meta.embedded.block.html` to match the `injectionSelector`?

Comment (mjbvz) [reply package.json]: I don't think so. `injectTo` specifies that target grammar id while the injectionSelector specifies the scope within the grammar to inject into

**Comment by mjbvz** [comment]
Thanks. Will publish update to extension shortly

**Pull request merged**